plugins/filesystem/local/adapter/adapter.php

```diff
@@ -326,4 +326,272 @@ private function getDate($date = null)
 
 		return $dateObj;
 	}
+
+	/**
+	 * Copies a file or folder to a destination
+	 * If the destination folder or file already exists, it will not overwrite them without
+	 * force.
+	 *
+	 * @param   string  $sourcePath       Source path of the file or directory
+	 * @param   string  $destinationPath  Destination path of the file or directory
+	 * @param   bool    $force            Set true to overwrite files or directories
+	 *
+	 * @return bool
+	 *
+	 * @since __DEPLOY_VERSION__
+	 * @throws MediaFileAdapterFilenotfoundexception
+	 */
+	public function copy($sourcePath, $destinationPath, $force = false)
+	{
+		// Get absolute paths from relative paths
+		$sourcePath = $this->rootPath . $sourcePath;
+		$destinationPath = $this->rootPath . $destinationPath;
+
+		if (!file_exists($sourcePath))
+		{
+			throw new MediaFileAdapterFilenotfoundexception;
+		}
+
+		// Check for existence of the file in destination
+		// if it does not exists simply copy source to destination
+
+		if (is_dir($sourcePath))
+		{
+			$success = $this->copyFolder($sourcePath, $destinationPath, $force);
+			return $success;
+		}
+		else
+		{
+			$success = $this->copyFile($sourcePath, $destinationPath, $force);
+			return $success;
+		}
+	}
+
+	/**
+	 * Copies a file
+	 *
+	 * @param   string  $sourcePath       Source path of the file or directory
+	 * @param   string  $destinationPath  Destination path of the file or directory
+	 * @param   bool    $force            Set true to overwrite files or directories
+	 *
+	 * @return bool
+	 *
+	 * @since __DEPLOY_VERSION__
+	 */
+	protected function copyFile($sourcePath, $destinationPath, $force = false)
```

Comment:
Set it to private, when there is a need we can always set it to protected. Going back will be harder.